src/unstructured_client/_hooks/custom/clean_server_url_hook.py

```diff
@@ -7,31 +7,61 @@
 from unstructured_client.httpclient import HttpClient
 
 
-class CleanServerUrlSDKInitHook(SDKInitHook):
-    """Hook fixing common mistakes by users in defining `server_url` in the unstructured-client"""
+def clean_server_url(base_url: str) -> str:
+    """Fix url scheme and remove the '/general/v0/general' path."""
+
+    if not base_url:
+        return ""
+    # -- add a url scheme if not present (urllib.parse does not work reliably without it)
+    if "http" not in base_url:
+        base_url = "http://" + base_url
+
+    parsed_url: ParseResult = urlparse(base_url)
+
+    if "api.unstructuredapp.io" in base_url:
+        if parsed_url.scheme != "https":
+            parsed_url = parsed_url._replace(scheme="https")
+
+    # -- path should always be empty
+    return urlunparse(parsed_url._replace(path=""))
 
-    def clean_server_url(self, base_url: str) -> str:
-        """Fix url scheme and remove the '/general/v0/general' path."""
 
-        if not base_url:
-            return ""
-        # -- add a url scheme if not present (urllib.parse does not work reliably without it)
-        if "http" not in base_url:
-            base_url = "http://" + base_url
+def choose_server_url(endpoint_url, client_url, default_endpoint_url) -> str:
+    """
+    Helper function to fix a breaking change in the SDK past 0.30.0.
+    When we merged the partition and platform specs, the client server_url stopped working,
+    and users need to pass it in the endpoint function.
+    For now, call this helper in the generated code to set the correct url.
+    """
 
-        parsed_url: ParseResult = urlparse(base_url)
+    # First, see if the endpoint has a url:
+    # s.general.partition(server_url=...)
+    if endpoint_url is not None:
```

Comment:
This means the endpoint_url is treated as an optional kwarg and it's currently not.

Reply:
This can come in from the generated code as None over [here](https://github.com/Unstructured-IO/unstructured-python-client/blob/a4bfd93eefbd909e1287f3d066f156b365b4a2e4/src/unstructured_client/general.py#L41)